### DISCUSSION TITLE
Connect to DB from external tool

### DISCUSSION THREAD

**Discussion by nivo33** [planetscale/discussion#148] [Q&A] [answered]
When trying to connect an external tool to planetscale (in this case www.metabase.com or www.retool.com), im trying to use the MySQL connectors. If I understand correctly, they should work.

However, to use SSL, they both require me to insert an SSL certificate (see image). However, since this is not running on my computer, I dont have the `etc/ssl/cert.pem` to direct them to. What SSL am I suppoed to be using?

<img width="732" alt="Screen Shot 2022-01-04 at 14 00 35" src="https://user-images.githubusercontent.com/19783311/148056023-de0b3588-9a83-413e-b87b-900a9f7deb74.png">

Thanks!

**Comment by sneakersgames** [comment]
I have the same problem. It allows me to connect but when I run a query I get"
```target: sporza.-.primary: vttablet: rpc error: code = InvalidArgument desc = Unknown or incorrect time zone: 'Europe/Brussels' (errno 1298) (sqlstate HY000) (CallerID: unsecure_grpc_client): Sql: "begin", BindVars: {}```

**Reply by leonardobork** [reply]
Same here! Help pls!

**Reply by jgreet** [reply]
Is this error in Metabase or Retool?

**Comment by jgreet** [comment, marked as answer]
In Metabase Cloud it looks like you can establish a secure connection use the following database settings:

- Pick MySQL
- Create a [connection string](https://docs.planetscale.com/concepts/connection-strings) in PlanetScale and use those details for Host, Database name, Username and Password
- Slide "Use a secure connection (SSL)?" to on
- For "Additional JDBC connection string options" add `sslMode=VERIFY_IDENTITY`

It does look like some basic functionality is broken once it's connected though. We'll take a look

**Reply by jgreet** [reply]
Looked in to the Metabase `No fields found` errors here https://github.com/planetscale/beta/discussions/161#discussioncomment-2161079

**Comment by rdvo** [comment]
How do you connec to planetscale from a remote nocode client liek Retool, Budibase, etc? 
Thanks

**Reply by jgreet** [reply]
The configuration will depend on the specific tool you're using but the goal with any of them is to enable SSL and certificate verification using the system's root certificate stores, same as [connecting your application](https://planetscale.com/docs/concepts/secure-connections).

For Budibase cloud, for example, I was able to use their MySQL connector after adding the `RejectUnauthorized: true` parameter like this:

<img width="702" alt="CleanShot 2022-12-12 at 14 30 37@2x" src="https://user-images.githubusercontent.com/391331/207137099-c3c4b3cf-ac54-4190-a89f-bec07633aae6.png">

See also https://github.com/Budibase/budibase/discussions/5515#discussioncomment-2614161

**Comment by dparrelli** [comment]
hey @jgreet - thanks for helping with this. Do you happen to know how to get it working with Retool? Can't seem to get it connected unfortunately. I receive an 'Unable to get local issuer certificate' error after selecting both the 'Verify CA Certificate' and 'Skip CA Certificate verification' options.
![image](https://user-images.githubusercontent.com/4967146/210334462-c973ae94-af0d-4a21-ab06-aea847c357ad.png)

**Reply by fooforge** [reply]
@dparrelli Retool appears to only support self-signed certificates, but you can work around this limitation and have Retool connect securely to your PlanetScale database by adding our certificate authorities' root certificate.

I am going to attach it as a `.txt` file on my reply. You will need to rename the file to `.pem`, though, as most 3rd party tools will not allow you to upload it otherwise.

The same certificate can also be downloaded from https://letsencrypt.org/certificates/, it's the ISRG Root X1.

Note that if we ever change our SSL provider, this certificate needs to be updated with the new provider's root certificate. If that happens, we'll make a note in [our changelog](https://planetscale.com/changelog).

[isrgx1.txt](https://github.com/planetscale/discussion/files/11294143/isrgx1.txt)

**Reply by dparrelli** [reply]
This is so incredibly helpful @fooforge and saved me a tonne of time/effort. I can confirm that it's working now 🙌

**Reply by jeroenransijn** [reply]
I am confused what to enter for the Client key and Client Cert fields. Any help appreciated.

**Reply by fooforge** [reply]
> I am confused what to enter for the Client key and Client Cert fields. Any help appreciated.

@jeroenransijn Looks like you have also opened a ticket. I've responded to it yesterday. In short: You'll leave these blank as they're not needed. You only have to save our certificate chain (or really only our CA's root certificate) to a file and drop that file in the _CA Certificates_ field in Retool.

**Reply by jeroenransijn** [reply]
Thanks! Ended up working.

**Reply by maravento** [reply]
ssl connection error: unknown error number
